src/Database/LSMTree/Internal/ByteString.hs

```diff
@@ -0,0 +1,49 @@
+{-# LANGUAGE BangPatterns #-}
+{-# LANGUAGE CPP          #-}
+-- | @bytestring@ extras
+module Database.LSMTree.Internal.ByteString (
+    shortByteStringFromTo,
+    byteArrayFromTo,
+) where
+
+import qualified Data.ByteString.Builder as BB
+import qualified Data.ByteString.Builder.Internal as BB
+import           Data.ByteString.Short (ShortByteString (SBS))
+import qualified Data.ByteString.Short.Internal as SBS
+import           Data.Primitive.ByteArray (ByteArray (..))
+import           Foreign.Ptr (minusPtr, plusPtr)
+
+-- | Copy of 'SBS.shortByteString', but with bounds (unchecked).
+--
+-- https://github.com/haskell/bytestring/issues/664
+{-# INLINE shortByteStringFromTo #-}
+shortByteStringFromTo :: Int -> Int -> ShortByteString -> BB.Builder
+shortByteStringFromTo = \i j sbs -> BB.builder $ shortByteStringCopyStepFromTo i j sbs
+
+-- | Like 'shortByteStringFromTo' but for 'ByteArray'
+--
+-- https://github.com/haskell/bytestring/issues/664
+--
+-- TODO: will break with newer bytestring
```

Comment:
How so? Isn't it still a `ByteArray` in `bytestring-0.12`?

Reply:
I haven't tried. There `SBS` compatibility pattern, but without trying, one cannot know